src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/TripScheduleImpl.java

```diff
@@ -18,27 +28,53 @@
      */
     private final int[] departures;
 
-    private final Trip originalTrip;
-
-    //TODO naming
-    private final TripPattern originalTripPattern;
-
-    private final int serviceCode;
+    /** How many seconds to shift the arrivals and departures (enabling reuse of arrival/departure arrays). */
+    private final int timeShift;
 
-    public TripScheduleImpl(int[] arrivals, int[] departures, Trip originalTrip, TripPattern originalTripPattern, int serviceCode) {
+    public TripScheduleImpl (
+            TripTimes originalTripTimes,
+            TripPattern originalTripPattern,
+            int[] arrivals,
+            int[] departures,
+            int timeShift
+    ) {
+        // Sanity check array and stop pattern dimensions.
+        final int numStops = originalTripTimes.getNumStops();
+        if (originalTripPattern.getStops().size() != numStops) {
+            LOG.error("TripPattern is not the same size as the TripTimes. This indicates a bug.");
+        }
+        if (arrivals.length != numStops) {
+            LOG.error("Arrivals arrays is not the same size as the TripTimes. This indicates a bug.");
+        }
+        if (departures.length != numStops) {
+            LOG.error("Departures arrays is not the same size as the TripTimes. This indicates a bug.");
+        }
+        this.originalTripTimes = originalTripTimes;
+        this.originalTripPattern = originalTripPattern;
         this.arrivals = arrivals;
         this.departures = departures;
-        this.originalTrip = originalTrip;
-        this.serviceCode = serviceCode;
-        this.originalTripPattern = originalTripPattern;
+        this.timeShift = timeShift;
```

Comment:
Right, OTP is already doing this time shifting and deduplication so now, since we're no longer duplicating all the arrays but instead reusing them, they remain deduplicated when used by the raptor code.